tensorflow/core/kernels/mkl_qmatmul_op_test.cc

```diff
@@ -0,0 +1,464 @@
+  AddInputFromArray<uint8>(kDummyShape, kDummyTensor);
+  AddInputFromArray<uint8>(kDummyShape, kDummyTensor);
+  AddInputFromArray<uint8>(kDummyShape, kDummyTensor);
+  AddInputFromArray<uint8>(kDummyShape, kDummyTensor);
+  AddInputFromArray<uint8>(kDummyShape, kDummyTensor);
+  AddInputFromArray<uint8>(kDummyShape, kDummyTensor);
+
+  TF_ASSERT_OK(RunOpKernel());
+  // Here are the results we expect, from hand calculations:
+  // (1 * 7) + (2 * 11) + (3 * 15) = 74
+  // (1 * 8) + (2 * 12) + (3 * 16) = 80
+  // (1 * 9) + (2 * 13) + (3 * 17) = 86
+  // (1 * 10) + (2 * 14) + (3 * 18) = 92
+  // (4 * 7) + (5 * 11) + (6 * 15) = 173
+  // (4 * 8) + (5 * 12) + (6 * 16) = 188
+  // (4 * 9) + (5 * 13) + (6 * 17) = 203
+  // (4 * 10) + (5 * 14) + (6 * 18) = 218
+  // Final result after Bias addition:
+  // 74+100=174, 80-200=-120, 86+300=386, 92-400=-308,
+  // 173+100=273, 188-200=-12, 203+300=503, 218-400=-182
+  Tensor expected(allocator(), DT_QINT32, TensorShape({2, 4}));
+  test::FillValues<qint32>(&expected,
+                           {174, -120, 386, -308, 273, -12, 503, -182});
+
+  const Tensor& output = *GetOutput(0);
+  const Tensor& mkl_shape_tensor = *GetOutput(3);
+  ConvMklToTF conv_comp;
+  Tensor output_quantized;
+  conv_comp.ConvertMKL2TF<qint32>(DT_QINT32, output, mkl_shape_tensor,
+                                  output_quantized);
+
+  test::ExpectTensorEqual<qint32>(expected, output_quantized);
+}
+
+// Two small matrices A of type uint8 (converted from signed integer)
+// and B of type int8  are multiplied and the result is added with float bias
+TEST_F(QuantizedMatMulTest, Small_WithNegInp) {
+  TF_ASSERT_OK(
+      NodeDefBuilder("quantized_mat_mul_op", "_MklQuantizedMatMulWithBias")
+          .Input(FakeInput(DT_QUINT8))
+          .Input(FakeInput(DT_QINT8))
+          .Input(FakeInput(DT_FLOAT))
+          .Input(FakeInput(DT_FLOAT))
+          .Input(FakeInput(DT_FLOAT))
+          .Input(FakeInput(DT_FLOAT))
+          .Input(FakeInput(DT_FLOAT))
+          .Input(FakeInput(DT_UINT8))  // MKL second tensor
+          .Input(FakeInput(DT_UINT8))  // MKL second tensor
+          .Input(FakeInput(DT_UINT8))  // MKL second tensor
+          .Input(FakeInput(DT_UINT8))  // MKL second tensor
+          .Input(FakeInput(DT_UINT8))  // MKL second tensor
+          .Input(FakeInput(DT_UINT8))  // MKL second tensor
+          .Input(FakeInput(DT_UINT8))  // MKL second tensor
+          .Attr("Toutput", DataTypeToEnum<qint32>::v())
+          .Attr("T", DataTypeToEnum<qint32>::v())
+          .Attr("input_quant_mode", "MIN_FIRST")
+          .Attr("_kernel", "QuantizedMklOp")
+          .Finalize(node_def()));
+  TF_ASSERT_OK(InitOp());
+  // The A matrix is:
+  // |  -1 |  -5 |  -9 |
+  // |  -2 |  -6 | -10 |
+  // |  -3 |  -7 | -11 |
+  // |  -4 |  -8 | -12 |
+  // The input array only contains unsigned bytes, so we specify the actual
+  // quantized values as Au8 = (Af32 - Min(A𝑓32) * Qa, where
```

Comment:
Missing one closing parenthesis.
```suggestion
  // quantized values as Au8 = (Af32 - Min(A𝑓32)) * Qa, where
```

Reply:
@penpornk Thank you very much. Please check the latest commit.